frontend/src/components/header/MenuOverlay.js

```diff
@@ -96,18 +97,21 @@ class MenuOverlay extends Component {
     if (targetValue) {
       this.setState({
         query: e.target.value,
+        pendingQuery: e.target.value ? true : false,
       });
 
       queryPathsRequest(e.target.value, 9)
         .then((response) => {
           this.setState({
```

Comment:
(not in requirements but as a scout boy rule)

While testing using Slow 3G i observed that 
* if i search for something => request(1) is sent to backend
* i don't wait to finish the request but i clear the search and write something else  => request(2) is sent to backend

In some cases if the the request(1) is finishing last, so my search is displaying the results of request(1) even though i would expect the results of request(2).

Suggestion to fix:
before updating the state, check if the query from the state is the one from our request, to make sure we get the right one.

Reply:
Added here too. But in this case things get more complicated as there is a debouncer used. So even with the changes done(using the suggestion) due to that debouncer we won't get the desired result. Exceeded the budget for the issue. Probably this should be tackled in a separate issue.